python/taichi/lang/expr.py

```diff
@@ -39,6 +41,20 @@ def __init__(self, *args, tb=None, dtype=None):
             self.ptr.set_tb(self.tb)
         self.ptr.type_check(impl.get_runtime().prog.config)
 
+    def __getitem__(self, indices):
```

Comment:
When calling matrix-matrix operations to build the program at Python frontend, we will use this to explicitly construct the indexing expression.

Reply:
Hmm I thought it was handled here? https://github.com/taichi-dev/taichi/blob/3677d934e3863fb74a7236dc3512ce2c6328b0fc/python/taichi/lang/ast/ast_transformer.py#L241-L250

Reply:
This is the case that it's handled by the ASTTransformer, but calling a library function won't be handled (because it's not a kernel, and it inlines the operations).

Reply:
1. Is it possible to make all library functions `ti.func`?
2. Are the changes in this file unrelated to this PR? If so shall we cancel them first?